compiler/next/test/frontend/testParseStringBytesLiterals.cpp

```diff
@@ -0,0 +1,182 @@
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "chpl/frontend/Parser.h"
+#include "chpl/queries/Context.h"
+#include "chpl/uast/BytesLiteral.h"
+#include "chpl/uast/CStringLiteral.h"
+#include "chpl/uast/Module.h"
+#include "chpl/uast/StringLiteral.h"
+#include "chpl/uast/Variable.h"
+
+// always check assertions in this test
+#ifdef NDEBUG
+#undef NDEBUG
+#endif
+
+#include <cassert>
+
+using namespace chpl;
+using namespace uast;
+using namespace frontend;
+
+static void testStringLiteral(Parser* parser,
+                              const char* testname,
+                              const char* str,
+                              StringLiteral::QuoteStyle expectQuoteStyle,
+                              const std::string& expectValue) {
+  std::string toparse = "var x = ";
+  toparse += str;
+  toparse += ";\n";
+  auto parseResult = parser->parseString(testname, toparse.c_str());
+  assert(parseResult.errors.size() == 0);
+  assert(parseResult.topLevelExpressions.size() == 1);
+  assert(parseResult.topLevelExpressions[0]->isModule());
+  auto module = parseResult.topLevelExpressions[0]->toModule();
+  assert(module->numStmts() == 1);
+  auto variable = module->stmt(0)->toVariable();
+  assert(variable);
+  auto strLit = variable->initExpression()->toStringLiteral();
+  assert(strLit);
+  assert(strLit->quoteStyle() == expectQuoteStyle);
+  assert(strLit->str() == expectValue);
+}
+static void testBytesLiteral(Parser* parser,
+                             const char* testname,
+                             const char* str,
+                             StringLiteral::QuoteStyle expectQuoteStyle,
+                             const std::string& expectValue) {
+  std::string toparse = "var x = b";
+  toparse += str;
+  toparse += ";\n";
+  auto parseResult = parser->parseString(testname, toparse.c_str());
+  assert(parseResult.errors.size() == 0);
+  assert(parseResult.topLevelExpressions.size() == 1);
+  assert(parseResult.topLevelExpressions[0]->isModule());
+  auto module = parseResult.topLevelExpressions[0]->toModule();
+  assert(module->numStmts() == 1);
+  auto variable = module->stmt(0)->toVariable();
+  assert(variable);
+  auto bytesLit = variable->initExpression()->toBytesLiteral();
+  assert(bytesLit);
+  assert(bytesLit->quoteStyle() == expectQuoteStyle);
+  assert(bytesLit->str() == expectValue);
+}
+static void testCStringLiteral(Parser* parser,
```

Comment:
There's a lot of code duplication among these three testXLiteral functions. I'd be tempted to try and factor the common code out or use templates.